modules/solid_mechanics/include/materials/NewtonIterationSingleVariable.h

```diff
@@ -0,0 +1,214 @@
+/****************************************************************/
+/* MOOSE - Multiphysics Object Oriented Simulation Environment  */
+/*                                                              */
+/*          All contents are licensed under LGPL V2.1           */
+/*             See LICENSE for full restrictions                */
+/****************************************************************/
+#ifndef NEWTONITERATIONSINGLEVARIABLE_H
+#define NEWTONITERATIONSINGLEVARIABLE_H
+
+#include "InputParameters.h"
+#include "ConsoleStream.h"
+
+/**
+ * Base class that provides capability for Newton return mapping
+ * iterations on a single variable
+ */
+
+class NewtonIterationSingleVariable;
```

Comment:
This name is a bit too general for something that requires a _trial stress_ in my opinion.